Platforms/interface/NativeWindowHandle.h

```diff
@@ -32,33 +32,63 @@
 #if PLATFORM_WIN32
 
 #    include "../Win32/interface/Win32NativeWindowHandle.hpp"
+
+DILIGENT_BEGIN_NAMESPACE(Diligent)
+
 typedef struct Win32NativeWindowHandle NativeWindowHandle;
 
+DILIGENT_END_NAMESPACE // namespace Diligent
```

Comment:
Ideally, I want to get rid of these duplicate namespace declaration in this file. Maybe I should just move the typedef to the platform specific file and just include it here?

Reply:
Ah, OK. Note that the rest of the platform-specific classes not being in a namespace is a mistake which I am planning to fix at some point.
I would structure the file as follows (and then fix the other headers similar later):

```cpp

#if PLATFORM_WIN32
#    include "../Android/interface/Win32NativeWindow.h"
...
#endif

DILIGENT_BEGIN_NAMESPACE(Diligent)

#if PLATFORM_WIN32
typedef struct Win32NativeWindow NativeWindow;
...
#endif

DILIGENT_END_NAMESPACE
```

There will be duplicated `#if-#elseif` struture, but this is probably a better way then having  multiple `BEGIN_NAMESPACE`/`END_NAMESPACE` sections